Directory.Build.props

```diff
@@ -6,6 +6,7 @@
     <IsClientLibrary Condition="'$(IsClientLibrary)' == '' and $(MSBuildProjectName.StartsWith('Azure.'))">true</IsClientLibrary>
     <IsDataPlaneProject Condition="'$(IsDataPlaneProject)' == '' and '$(IsClientLibrary)' == 'true'">true</IsDataPlaneProject>
     <IsDataPlaneProject Condition="'$(IsDataPlaneProject)' == '' and $(MSBuildProjectDirectory.Contains('data-plane'))">true</IsDataPlaneProject>
+    <IsDataPlaneProject Condition="'$(IsDataPlaneProject)' == '' and $([System.Text.RegularExpressions.Regex]::IsMatch('$(MSBuildProjectDirectory)', '.*(\\sdk\\.*?\\Microsoft.Azure..*?\\)'))">true</IsDataPlaneProject>
```

Comment:
No need to use the regex as that isn't as efficient just make this condition `$(MSBuildProjectName.StartsWith('Microsoft.Azure.'))`

Reply:
We will likely also need to add some exclusions for management libraries.

Reply:
Ok, that why I used the regex instead. I will try to get it working that way.

Reply:
That regex will not work once you move the management libraries under sdk as well. I suggest simply changing the conditions to look like once we need to differentiate the management projects as well. 

```
<IsMgmtProject Condition="'$(IsMgmtProject)' == '' and '$(MSBuildProjectName.Contains('.Management.'))>true<IsMgmtProject>
<IsDataPlaneProject Condition="'$(IsDataPlanProject)' == '' and $(MSBuildProjectName.StartsWith('Microsoft.Azure.')) and '$(IsMgmtProject)' != 'true'>true</IsDataPlaneProject>
```

Reply:
Do I also need to include the new directory in data-plane.proj

Reply:
No I already included it in there when I moved the template so it should work.